egs/wsj/s5/steps/dict/prune_pron_candidates.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+
+# Copyright 2016  Xiaohui Zhang
+# Apache 2.0.
+
+from __future__ import print_function
+from collections import defaultdict
+import argparse
+import sys
+import math
+
+def GetArgs():
+    parser = argparse.ArgumentParser(description = "Prune pronunciation candidates based on soft-counts from lattice-alignment"
+                                     "outputs, and a reference lexicon. Basically, for each word we sort all pronunciation"
+                                     "cadidates according to their soft-counts, and then select the top r * N candidates"
+                                     "(For words in the reference lexicon, N = # pron variants given by the reference"
+                                     "lexicon; For oov words, N = avg. # pron variants per word in the reference lexicon)."
+                                     "r is a user-specified constant, like 2.",
+                                     epilog = "See steps/dict/learn_lexicon.sh for example")
+
+    parser.add_argument("--r", type = float, default = "2.0",
+                        help = "a user-specified ratio parameter which determines how many"
+                        "pronunciation candidates we want to keep for each word.")
+    parser.add_argument("pron_stats", metavar = "<pron-stats>", type = str,
+                        help = "File containing soft-counts of all pronounciation candidates; "
+                        "each line must be <soft-counts> <word> <phones>")
+    parser.add_argument("ref_lexicon", metavar = "<ref-lexicon>", type = str,
+                        help = "Reference lexicon file, where we obtain # pron variants for"
+                        "each word, based on which we prune the pron candidates.")
+    parser.add_argument("pruned_prons", metavar = "<pruned-prons>", type = str,
+                        help = "A file in lexicon format, which contains prons we want to" 
+                        "prune away from the pron_stats file.")
```

Comment:
I think what you mean here is the prons we want to retain from the pron_stats file.
When you prune a tree, the parts that are pruned are the bits you cut off from the tree.